### ISSUE TITLE
Sorting PDS members in the descending order causes unwanted behavior

### ISSUE THREAD

**Issue by adam-wolfe** [zowe/vscode-extension-for-zowe#2536]
**Describe the bug**

When PDS members are sorted by date modified or user ID in the descending order, when the PDS is collapsed and re-expanded, the order of items with identical search values (i.e., same date last modified or same user ID) reverses (see screen recording)

**To Reproduce**

1. Sort PDS members by user ID or date modified in descending order
2. Collapse the PDS
3. Expand the PDS
4. Notice that the order of items with the same user id or date modified has reversed

**Expected behavior**

The order should not change when the PDS is collapsed and re-expanded.

**Screenshots**
Sorry for the huge gif.
![SortDescendingIssue](https://github.com/zowe/vscode-extension-for-zowe/assets/122040687/8f46cd1e-93e8-430a-8f47-f8e17f9adced)

**Desktop (please complete the following information):**

- OS: Windows 10
- Zowe Explorer Version: 
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

This does not happen when the order is set to ascending.

**Comment by zFernand0** [comment]
Is this due to the fact that we do not sort down to the millisecond in which something was created?
If we want, we could introduce a second sort criteria (e.g. `name` -when sorting by date, `date`-when sorting by name/user)

**Comment by traeok** [comment]
> Is this due to the fact that we do not sort down to the millisecond in which something was created? If we want, we could introduce a second sort criteria (e.g. `name` -when sorting by date, `date`-when sorting by name/user)

Technically we do sort by the smallest unit, as the date comparison was `dateA < dateB`. I've decided to adjust this slightly by using Day.js to compare the dates down to the very second, as it is the smallest unit of time that z/OSMF provides from their API responses.

But I agree, I think a user would expect it to fall back to sorting by name when the values are equal. Since it didn't take long to clean up the sort function, and the updated function is easier to read, I decided to address that in #2538
